tests/unit/RecordUnitTests.d

```diff
@@ -524,3 +524,77 @@ D");
 
 }
 
+@("issue #166 - translate record with name from macro")
+unittest {
+    assertTranslates(
+    q{
+
+#define __FSID_T_TYPE struct { int __val[2]; }
+typedef  __FSID_T_TYPE __fsid_t;
+typedef __fsid_t fsid_t;
+
+},
+    q{extern (C):
+
+struct __fsid_t
+{
+    int[2] __val;
+}
+
+alias fsid_t = __fsid_t;
+
+});
+
+}
+
+
+@("issue #166 - translate record with name from macro in a literal macro")
+unittest {
+    assertTranslates(
+        q{
```

Comment:
q"C [...]"  q"D [...]" keep it consistent.

Reply:
Right, forgot.